app/goal_routes.py

```diff
@@ -0,0 +1,132 @@
+from flask import Blueprint, request, make_response, abort, jsonify
+from app.models.goal import Goal
+from app.models.task import Task
+from app import db
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix = "/goals")
+
+# HELPER FUNCTION
+def validate_id(class_obj, id):
+    try:
+        object_id = int(id)
+    except:
+        abort(make_response({"details": f"{class_obj} {id} is invalid"}, 400))
+
+    query_result = class_obj.query.get(object_id)
+    if not query_result:
+        abort(make_response({"details": f"{class_obj} {id} is not found"}, 404))
+    
+    return query_result
+
+# CREATE RESOURCE
+@goals_bp.route("", methods = ["POST"])
+def create_goal():
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal.from_json(request_body)
+    except KeyError:
+        return make_response({"details": "Invalid data"}, 400)
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return make_response(new_goal.to_dict(), 201)
+
+# GET ALL RESOURCE
+@goals_bp.route("", methods = ["GET"])
```

Comment:
```suggestion
@goals_bp.route("", methods=["GET"])
```

Reply:
I won't duplicate this suggestion any further. But consider this comment for all following the route signatures even if it isn't explicitly corrected